File: src/Dinder.Domain/Entities/Message.cs
namespace Dinder.Domain.Entities;

public sealed class Message
{
    public Guid Id { get; private set; }
    public Guid ConversationId { get; private set; }
    public Guid SenderId { get; private set; }
    public string Content { get; private set; }
    public DateTime SentAt { get; private set; }
    public DateTime? ReadAt { get; private set; }

#pragma warning disable CS8618
    private Message() { } // EF Core
#pragma warning restore CS8618

    public Message(Guid conversationId, Guid senderId, string content)
    {
        Id = Guid.NewGuid();
        ConversationId = conversationId;
        SenderId = senderId;
        Content = content;
        SentAt = DateTime.UtcNow;
    }

    public void MarkRead()
    {
        ReadAt = DateTime.UtcNow;
    }
}

File: mobile/app/src/main/java/com/dinder/app/domain/model/Message.kt
package com.dinder.app.domain.model

/**
 * Domain model representing a chat message.
 */
data class Message(
    val messageId: String,
    val conversationId: String,
    val senderId: String,
    val content: String,
    val sentAt: String,
    val readAt: String? = null
)
